src/target/riscv/riscv.c

```diff
@@ -852,24 +862,35 @@ static int maybe_add_trigger_t2_t6(struct target *target,
 		struct trigger_request_info lt_1 = {
 			.tdata1 = fields.common | fields.size.any | fields.chain.enable |
 				fields.match.lt,
-			.tdata2 = trigger->address,
+			.tdata2 = trigger->address + trigger->length,
 			.tdata1_ignore_mask = fields.tdata1_ignore_mask
 		};
 		struct trigger_request_info ge_2 = {
 			.tdata1 = fields.common | fields.size.any | fields.chain.disable |
 				fields.match.ge,
-			.tdata2 = trigger->address + trigger->length,
+			.tdata2 = trigger->address,
 			.tdata1_ignore_mask = fields.tdata1_ignore_mask
 		};
 		ret = try_setup_chained_match_triggers(target, trigger, lt_1, ge_2);
 		if (ret != ERROR_TARGET_RESOURCE_NOT_AVAILABLE)
 			return ret;
 	}
 	LOG_TARGET_DEBUG(target, "trying to setup equality match trigger");
+	if (trigger->length > 1) {
+		/* This message is displayed only once per session to avoid
+		 * overwhelming the user with such messages on resume.
+		 */
+		static bool warning_was_displayed;
```

Comment:
Please, do not use static local variables - because that's in fact almost a global variable, just hidden inside a function :-)

My recommendation would be to: 
- Add a bool flag to riscv013_info_t
- Print once a more generic warning message. Just say that it was not possible to add a trigger for range of addresses, and some triggers may match only on the beginning of the respective range. No need to mention concrete addresses in this generic warning.
- Turn the current "warning" message to "debug" message and print it every time. It can be very useful for troubleshooting.

Reply:
Addressed.